### DISCUSSION TITLE
Minor Version firmware - whats meant by that?

### DISCUSSION THREAD

**Discussion by andrematzke** [arendst/Tasmota#17438] [Q&A] [answered]
in the Readme.md its recommended not flashing from a minor to a minor version, and recommended flash a full version first and then its minor version.

its a pitty that its not explained, whats meant by minor version, and what exactly by full version.. since there is e.g. something like that: 9.3.1 ?
is the x.y.1 the minor ?
and x the full or x.y = x.y.0 the full ? or x.y.0 => x.y = minor ?

I am confused.

And how to reach the "full versions" if on the release-download pages mostly minor versions are offered ? 
Do I have to try out guessing the url like that:
http://ota.tasmota.com/tasmota/release-9.3.0/  (which gives me in this example a 404)

I am really confused ? Can someone give me a hint ?

TIA,
Andre

**Comment by sfromis** [comment, marked as answer]
What I see in the readme, does not talk about "minor versions", but about "minimal versions", which is unrelated to the numbers. The `tasmota-minimal.bin.gz` binary for ESP8266 is a special binary for the exclusive purpose of being a stepping stone during upgrades. This means that it is unable to change settings any way, while the full binaries (everything but minimal) is able to upgrade settings from one version to the next. Due to this, it is pointless and detrimental to the upgrade process if you upgrade from one minimal to the next, as you'll lose out on the important upgrades to the settings data structure.

During the upgrade steps, you need to have been on a full Tasmota for every of the recommended steps. In most cases, the easiest way is to start by upgrading to `tasmota-lite` on the old version, do the upgrade stepping through tasmota-lite versions, and then upgrading back to full when having gotten to the target (newest) version. This way, you can avoid using the minimal binaries, which reduces the risk of irrecoverable errors if something does not work out.

You need not guess URL's, the recommended steps and links are available on
https://tasmota.github.io/docs/Upgrading/#upgrade-flow

**Reply by andrematzke** [reply]
thx. this was very helpful !
